packages/vite/src/node/ssr/ssrExternal.ts

```diff
@@ -151,46 +151,56 @@ export function createIsConfiguredAsSsrExternal(
   }
 }
 
-function createIsSsrExternal(
-  config: ResolvedConfig
-): (id: string) => boolean | undefined {
-  const processedIds = new Map<string, boolean | undefined>()
-
-  const { ssr, root } = config
+function createIsSsrExternal(config: ResolvedConfig): (id: string) => boolean {
+  const processedIds = new Map<string, boolean>()
 
   const isConfiguredAsExternal = createIsConfiguredAsSsrExternal(config)
 
   const resolveOptions: InternalResolveOptions = {
-    root,
+    root: config.root,
     preserveSymlinks: config.resolve.preserveSymlinks,
     isProduction: false,
     isBuild: true
   }
 
-  const isExternalizable = (id: string) => {
-    if (!bareImportRE.test(id) || id.includes('\0')) {
-      return false
-    }
-    return !!tryNodeResolve(
+  const resolve = (id: string) => {
+    return tryNodeResolve(
       id,
       undefined,
       resolveOptions,
-      ssr?.target === 'webworker',
+      config.ssr?.target === 'webworker',
       undefined,
       true,
-      true // try to externalize, will return undefined if not possible
+      false
```

Comment:
We need to pass true here or the branch testing deep imports will not be evaluated

Reply:
It is also filtering .css files when passing true, we need more tests in ssr-deps 🤔

Reply:
Hmm I changed to true so we can resolve to the full path to analyze `node_modules` though. I'll try to find another way for this.

Reply:
You could do these checks inside tryNodeResolve when passing true

Reply:
Thanks for the hint. Now the fix is a lot simpler than before 😅

Reply:
Nice! I think we still need a refactoring to make the branches more clear, we can get there in 3.1